apps/server/src/modules/tool/context-external-tool/controller/api-test/tool-context.api.spec.ts

```diff
@@ -407,4 +410,68 @@ describe('ToolContextController (API)', () => {
 			});
 		});
 	});
+
+	describe('[GET] tools/context-external-tools/:contextExternalToolId', () => {
+		const setup = async () => {
+			const school: School = schoolFactory.buildWithId();
+
+			const { teacherUser, teacherAccount } = UserAndAccountTestFactory.buildTeacher({ school }, [
+				Permission.CONTEXT_TOOL_ADMIN,
+			]);
+
+			const course: Course = courseFactory.buildWithId({
+				teachers: [teacherUser],
+				school,
+			});
+
+			const externalTool: ExternalToolEntity = externalToolEntityFactory.buildWithId();
+			const schoolExternalTool: SchoolExternalToolEntity = schoolExternalToolEntityFactory.buildWithId({
+				school,
+				tool: externalTool,
+				toolVersion: 1,
+			});
+			const contextExternalTool: ContextExternalToolEntity = contextExternalToolEntityFactory.buildWithId({
+				contextId: course.id,
+				schoolTool: schoolExternalTool,
+				toolVersion: 1,
+				contextType: ContextExternalToolType.COURSE,
+			});
+
+			await em.persistAndFlush([
+				school,
+				course,
+				externalTool,
+				schoolExternalTool,
+				contextExternalTool,
+				teacherAccount,
+				teacherUser,
+			]);
+			em.clear();
+
+			currentUser = mapUserToCurrentUser(teacherUser);
+
+			return {
+				contextExternalTool,
+				schoolExternalTool,
+			};
+		};
+
+		describe('when contextExternalToolId, contextType and contextId is given', () => {
```

Comment:
what happpens when some info of these is not given? Should there be a test for a non-happy case?